src/headwordsmodel.cc

```diff
@@ -80,21 +80,19 @@ void HeadwordListModel::appendWord( const QString & word )
 void HeadwordListModel::requestFinished()
 {
   // See how many new requests have finished, and if we have any new results
-  for ( auto i = queuedRequests.begin(); i != queuedRequests.end(); ) {
-    if ( ( *i )->isFinished() ) {
-      if ( !( *i )->getErrorString().isEmpty() ) {
-        qDebug() << "error:" << ( *i )->getErrorString();
+  auto snapshot = queuedRequests.snapshot();
+  for ( const auto & request : snapshot ) {
+    if ( request->isFinished() ) {
+      if ( !request->getErrorString().isEmpty() ) {
+        qDebug() << "error:" << request->getErrorString();
       }
-      else if ( ( *i )->matchesCount() ) {
-        auto allmatches = ( *i )->getAllMatches();
+      else if ( request->matchesCount() ) {
+        auto allmatches = request->getAllMatches();
         for ( auto & match : allmatches ) {
           filterWords.append( QString::fromStdU32String( match.word ) );
         }
       }
-      queuedRequests.erase( i++ );
-    }
-    else {
-      ++i;
+      queuedRequests.remove( request );
```

Comment:
`erase` and `remove` are not the same.

`remove` take a value and find equivalents in the whole list

`erase` take an iterator and delete the item at the position

Reply:
`erase` work together with iterator loop.

`remove` used with `for range` loop.   as the snapshot is only a copy of original list.   erase do not work here.

Reply:
![image](https://github.com/user-attachments/assets/aa6e6323-82bd-463e-8a0e-fe3dcd6c3812)